src/Analyser/NodeScopeResolver.php

```diff
@@ -2572,6 +2575,9 @@ static function (Node $node, Scope $scope) use ($nodeCallback): void {
 				$throwPoints = array_merge($throwPoints, $armResult->getThrowPoints());
 				$matchScope = $matchScope->filterByFalseyValue($filteringExpr);
 			}
+			if (!$hasDefaultCond) {
+				$throwPoints[] = ThrowPoint::createExplicit($scope, new ObjectType(UnhandledMatchError::class), $expr, false);
```

Comment:
This isn't sufficient. It's possible that the match handles all the cases and doesn't throw the exception, even without the default cause.

See the rule on how I figure that out: https://github.com/phpstan/phpstan-src/blob/d0ab72ac815f4a8716fa110d5c1e6caba6ceeb47/src/Rules/Comparison/MatchExpressionRule.php#L82-L91

Additionally, I'd like the linked error to not be reported if the match is surrounded by a try-catch for `UnhandledMatchError`.

Reply:
Thanks! I wasn't aware of this rule. I'll try to fix it sometime this week.